hack/release-helper.sh

```diff
@@ -0,0 +1,257 @@
+#   https://github.com/confidential-containers/confidential-containers/blob/main/.github/ISSUE_TEMPLATE/release-check-list.md
+# The script does not automate git commits or pushes, and it does not open PRs.
+#
+# Note: This script intentionally uses sed. yq (or even python libraries like
+# "yaml" or "ruamel.yaml") do not preserve yaml formatting.
+#
+
+
+
+
+# Assumption: This script lives in operator's top-level hack/ folder
+script_dir=$(dirname "$(readlink -f "$0")")
+proj_root=$(readlink -f ${script_dir}/..)
+
+
+
+
+function usage_and_exit() {
+    echo
+    echo "Usage:"
+    echo "  $0  <ACTION> [ARGS]"
+    echo
+    echo "  ACTION"
+    echo "  One of { check, update }"
+    echo "  check   Check that it's OK to proceed with the release. May modify"
+    echo "          local files if needed but does not commit or push them."
+    echo "          Note: Takes no args."
+    echo "  update  Update the operator repo with new tags. Modifies local"
+    echo "          files but does not commit or push them."
+    echo "          Note: All args are required"
+    echo
+    echo "  ARGS"
+    echo "  -o  The operator tag for your release"
+    echo "      Example: v0.9.0"
+    echo "  -e  The enclave-cc tag for your release"
+    echo "      Example: v0.9.1"
+    echo "  -k  The kata tag for your release (n.b. no 'v' prefix)"
+    echo "      Example: 3.6.0"
+    echo
+    echo "Example usage:"
+    echo "    $0 check"
+    echo "    $0 update -o v0.9.0 -e v0.9.1 -k 3.6.0"
+    echo
+    exit 1
+}
+
+
+function bail_if_dirty_repo() {
+    git diff --quiet ||
+          (echo
+           echo "Error: Must use a clean repo to use this script."
+           echo "       Stash, commit, branch, or use a fresh clone, etc."
+           echo
+           exit 1)
+}
+
+
+function parse_args() {
+    # pull the ACTION from the args list
+    if [ $# = 0 ]; then
+        usage_and_exit
+    fi
+    action=$1
+    shift
+
+    # parse the o, e, and k args
+    while getopts ":o:e:k:" opt; do
+        case "${opt}" in
+            o)
+                operator_tag=${OPTARG} ;;
+            e)
+                enclave_cc_tag=${OPTARG} ;;
+            k)
+                kata_tag=${OPTARG} ;;
+            *)
+                usage_and_exit ;;
+        esac
+    done
+
+    # minimal checking: ensure all args are provided when using 'update'
+    if [ ${action} = "update" ]; then
+        if [[ ! -v operator_tag ]] || [[ ! -v enclave_cc_tag ]] || [[ ! -v kata_tag ]]; then
+            echo "Error: Missing ARGS"
+            usage_and_exit
+        fi
+    fi
+}
+
+
+#
+# Make sure that the pre-reqs container ships the same version of the nydus
+# snapshotter that we use in the kata ci. Update the Makefile if needed.
+#
+function check_nydus() {
+    echo "Checking that the operator's nydus version matches the one used in kata"
+
+    kata_versions="https://raw.githubusercontent.com/kata-containers/kata-containers/main/versions.yaml"
```

Comment:
Hi @portersrc !

I think that we will need to fetch versions.yaml from the same kata version used on the operator release, rather than `main`. Right?

Reply:
100%, thanks for catching this